### ISSUE TITLE
One of the opcodes GCC uses to implement the C modulo operator is broken.

### ISSUE THREAD

**Issue by snickerbockers** [washingtondc-emu/washingtondc#27]
The following disassembly is part of a new regression test I am trying to write.  Its purpose is to take the compute in register r0 modulo 15, and send that to printf as the first argument (register r5).

Something goes wrong, and the value ends up being 979785262.  This code works on real hardware, so there must be a problem with WashingtonDC.

The initial value of r0 is 0x5851f42d.  When printf gets called, r5 is 979785262.

```
   0x8c0102da:	mov.l	0x8c0103ac,r1	! 0x88888889
   0x8c0102dc:	mov	#4,r2
   0x8c0102de:	mov.l	0x8c0103b0,r4	! 0x8c047e64
   0x8c0102e0:	dmuls.l	r1,r0
   0x8c0102e2:	mov	#-3,r1
   0x8c0102e4:	sts	mach,r8
   0x8c0102e6:	add	r0,r8
   0x8c0102e8:	shad	r1,r8
   0x8c0102ea:	mov	r0,r1
   0x8c0102ec:	shll	r1
   0x8c0102ee:	subc	r1,r1
   0x8c0102f0:	sub	r1,r8
   0x8c0102f2:	mov	r8,r1
   0x8c0102f4:	shld	r2,r1
   0x8c0102f6:	sub	r8,r1
   0x8c0102f8:	mov	r0,r8
   0x8c0102fa:	mov.l	0x8c0103b4,r0	! 0x8c03d3a0 <printf>
   0x8c0102fc:	sub	r1,r8
   0x8c0102fe:	jsr	@r0
   0x8c010300:	mov	r8,r5
```

**Comment by snickerbockers** [comment]
I'm very confident that dmuls.l is the culprit here.

**Comment by snickerbockers** [comment]
This issue was fixed in commit 51892a945fefbf0cfe0855abef74494837483f97.
